src/interface/guide/components/SegmentedTimeline.tsx

```diff
@@ -0,0 +1,192 @@
+import { useFight } from 'interface/report/context/FightContext';
+
+export interface TimeWindow {
+  /** Start time of the window in milliseconds */
+  startTime: number;
+  /** End time of the window in milliseconds */
+  endTime: number;
+}
+
+export interface TimelineSegment {
+  /** Start time of the segment in milliseconds */
+  start: number;
+  /** End time of the segment in milliseconds */
+  end: number;
+  /** Color of the segment */
+  color: string;
+  /** Opacity of the segment (0-1) */
+  opacity?: number;
+  /** Tooltip label for the segment */
+  label: string;
+}
+
+export interface TimelineMarker {
+  /** Timestamp of the marker in milliseconds */
+  timestamp: number;
+  /** Tooltip label for the marker */
+  label: string;
+  /** Color of the marker (default: white) */
+  color?: string;
+}
+
+export interface SegmentedTimelineProps {
+  /** Optional windows to show on the timeline (defaults to full fight if not specified) */
+  windows?: TimeWindow[];
+  /** Segments to render on the timeline */
+  segments: TimelineSegment[];
+  /** Optional markers to show on the timeline */
+  markers?: TimelineMarker[];
+}
+
+const EMPTY_MARKERS: TimelineMarker[] = [];
+
+/**
+ * Renders a timeline visualization with colored segments and optional markers.
+ *
+ * This is a generalized timeline component that can visualize any time-based data
+ * across a fight or specific time windows. Common uses include:
+ * - Cooldown availability
+ * - Buff/debuff uptime
+ * - Resource availability
+ * - Phase indicators
+ *
+ * @param windows - Optional time windows to display (defaults to full fight)
+ * @param segments - Colored segments to render on the timeline
+ * @param markers - Optional markers (pins) to show on the timeline
+ */
+export default function SegmentedTimeline({
+  windows,
+  segments,
+  markers = EMPTY_MARKERS,
```

Comment:
nit: this can just be `[]`
```suggestion
  markers = [],
```